typing/layouts.mli

```diff
@@ -84,8 +84,10 @@ module Layout : sig
 
   (** This layout is the top of the layout lattice. All types have layout [any].
       But we cannot compile run-time manipulations of values of types with layout
-      [any]. *)
-  val any : t
+      [any].
+
+      If [missing_cmi_for] is [None], this value is guaranteed to be shared. *)
+  val any : missing_cmi_for:Path.t option -> t
```

Comment:
I'll stop short of suggesting a default argument, but maybe we should have `any_not_missing` or something - this gets pretty verbose in the code.